contrib/endpoints/src/api_manager/check_security_rules.cc

```diff
@@ -182,39 +129,38 @@ void AuthzChecker::Check(
               // If the parsing of the release body is successful, then call the
               // Test Api for firebase rules service.
               if (status.ok()) {
```

Comment:
That is another way but here is how that code will look.

if (status.ok()) {                
                checker->env_->LogDebug(            
                    std::string("GetReleasName succeeded with ") + body);
                status = checker->ParseReleaseResponse(body, &ruleset_id);  
                  if (status.ok()) {
                       checker->request_handler_ = std::unique_ptr<FirebaseRequest>(
                    new FirebaseRequest(ruleset_id, checker->env_, context));
                checker->CallNextRequest(final_continuation);     
                    }  else { // I cannot remove this else statement since this will not go into the else part below
                          final_continuation(status)
                     }                                                                                                      
              } else {
                checker->env_->LogError(std::string("GetReleaseName for ") +
                                        GetReleaseUrl(*context.get()) + 
                                        " with status " + status.ToString());
                status = Status(Code::INTERNAL, kFailedFirebaseReleaseFetch);                                                                                                       
              }
         
The code does not look as good with status checks within status checks. Which is why I have written code in the way it is.